torch_xla/csrc/lowering_context.cpp

```diff
@@ -52,24 +56,90 @@ class HloMetadataSetter {
         absl::StrReplaceAll(node->op().ToString(), {{":", "_"}});
     metadata.set_op_type(op_type);
     const torch::lazy::MetaData& nmeta = node->metadata();
-    std::string op_name_prefix;
-    if (!nmeta.scope.empty()) {
-      op_name_prefix =
-          absl::StrCat(absl::StrReplaceAll(nmeta.scope, {{":", "_"}}), "/");
+
+    ExtendedFrameInfo* efi = nullptr;
+
+    if (FLAGS_torch_lazy_ir_debug) {
+      TF_VLOG(1) << "PopulateXlaOpMetadata: Debug lowering enabled";
+      efi = static_cast<ExtendedFrameInfo*>(node->user_metadata());
```

Comment:
dynamic_cast

Reply:
This was intentional since the if the code segments here and ir.cpp match the static cast should always succeed.  The aim was to avoid dynamic cast runtime costs from referencing RTTI.